x-pack/metricbeat/module/azure/azure.go

```diff
@@ -74,20 +109,37 @@ func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
 		config.Resources = resources
 	}
 	// instantiate monitor client
-	monitorClient, err := NewClient(config)
-	if err != nil {
-		return nil, fmt.Errorf("error initializing the monitor client: module azure - %s metricset: %w", metricsetName, err)
+	var monitorClient *Client
+	var monitorBatchClient *BatchClient
+	if containsString(monitorMetricsets, metricsetName) && config.EnableBatchApi {
```

Comment:
We use the batch API components only if:

1. the metricset is supported
2. batch api is enables

right?

Reply:
exactly. if the metricset name is part of the supported (monitor metricset and the ones that user monitor under the hood) and the parameter is enabled.

Reply:
Can we use a name that conveys this idea, like `supportedMonitorMetricsets`, for clarity?